File: android/src/main/kotlin/com/example/scgateway_flutter_plugin/models/SmallcaseStatsDTO.kt
package com.example.scgateway_flutter_plugin.models

import com.google.gson.annotations.SerializedName

data class SmallcaseStatsDTO(

        @SerializedName("returns")
        val returns: SmallcaseReturns? = null,

        @SerializedName("indexValue")
        val indexValue: Double? = null,

        @SerializedName("unadjustedValue")
        val unadjustedValue: Double? = null,

        @SerializedName("divReturns")
        val divReturns: Double? = null,

        @SerializedName("lastCloseIndex")
        val lastCloseIndex: Double? = null,

        @SerializedName("minInvestAmount")
        val minInvestAmount: Double? = null,

        @SerializedName("ratios")
        val ratios: RatiosDTO? = null
)

File: example/lib/app/models/SmallcaseStatsDTO.dart
import 'dart:ffi';

import 'package:scgateway_flutter_plugin_example/app/models/RatiosDTO.dart';
import 'package:scgateway_flutter_plugin_example/app/models/SmallcaseReturns.dart';

class SmallcaseStatsDTO {

  final SmallcaseReturns returns;

  final num indexValue;

  final num unadjustedValue;

  final dynamic divReturns;

  final num lastCloseIndex;

  final dynamic minInvestAmount;

  final RatiosDTO ratios;

  SmallcaseStatsDTO({
    required this.returns,
    required this.indexValue,
    required this.unadjustedValue,
    this.divReturns,
    required this.lastCloseIndex,
    this.minInvestAmount,
    required this.ratios
});

  factory SmallcaseStatsDTO.fromJson(Map<String, dynamic> parsedJson) {

    return SmallcaseStatsDTO(
      returns: SmallcaseReturns.fromJson(parsedJson['returns']),
      indexValue: parsedJson['indexValue'],
      unadjustedValue: parsedJson['unadjustedValue'],
      divReturns: parsedJson['divReturns'],
      lastCloseIndex: parsedJson['lastCloseIndex'],
      minInvestAmount: parsedJson['minInvestAmount'],
      ratios: RatiosDTO.fromJson(parsedJson['ratios'])
    );

  }

}